### PR TITLE
Example with a progress bar while loading

### PR DESCRIPTION
**Pull request by vorner**
Result of discussion on #579.

This adds another example, one that pretends to load something and updates a progress bar as it goes.

## Checks

- [X] I have read `CONTRIBUTING.md`.
- [X] I have updated `CHANGES.md`, with [BREAKING] next to all breaking changes
- [X] I have updated the documentation if necessary

(the example is a form of documentation on its own)

### CODE REVIEW

**Comment by ryanisaacg** [comment]
Probably don't want to use `sleep` here, see #580.

**Comment by lenscas** [comment]
suggestion to replace the sleep with, load a tiny image a lot of times? That would probably also look more real as you probably want the progress bar for loading assets.

**Comment by vorner** [comment]
Hmm. I wanted something that:
* Takes some predictable time, so the bar is visible when people run the example.
* Doesn't introduce unrelated concepts/parts of the library, as that would be distracting and probably should go into its own example.

Loading bunch of assets kind of doesn't satisfy either ☹.

I could implement a simple busy-loop there instead, which should work on the web too. Or would you still prefer loading some images instead, even though these might be unpredictably slow/fast and „clutter“ the example? How would loading a single image behave on the web million times, would it download it million times too?

**Comment by ryanisaacg** [comment]
I don't think hammering the user's client with an artificial workload is a good plan. I would be fine with throwing a TODO on the code and linking to it from #580.

**Comment by vorner** [comment]
OK, I've added a comment there with a TODO and a link (it's the only change since the previous version)

File: examples/06_loading_screen.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        Color::YELLOW,
+    );
+
+    let width = 700.0 * progress as f32 / total as f32;
+    gfx.fill_rect(
+        &Rectangle::new(Vector::new(50.0, 500.0), Vector::new(width, 25.0)),
+        Color::BLUE,
+    );
+
+    // Might as well draw some image above the bar too...
```
Comment (ryanisaacg) [inline examples/06_loading_screen.rs]: Do you plan to come back to this or should we remove it and merge?

Comment (vorner) [reply examples/06_loading_screen.rs]: I'm not sure I understand. Remove what, exactly? The comment is more like an suggestion for the reader than TODO for myself, that the progress bar in itself doesn't look very appealing, but adding an image there might help. Or is it confusing? If so, it's probably fine removing, or I can try rewording it.

Comment (ryanisaacg) [reply examples/06_loading_screen.rs]: Oh, I see. We might want to add 'In a real application' before 'Might' so the reader knows they're being addressed.

Comment (vorner) [reply examples/06_loading_screen.rs]: OK, I've tried to reword it a bit so its clearer.

**Comment by vorner** [comment]
Could I ask if there's anything this still waits for?

**Comment by ryanisaacg** [comment]
Sorry, things are tumultuous right now! I'll merge this.

**Pull request merged**